modules/authentication/proc-enabling-authentication-with-your-custom-authentication-provider.adoc

```diff
@@ -0,0 +1,212 @@
++
+.Secrets for the LDAP example
+====
+LDAP_URL:: Your LDAP server URL, such as `ldap://_<my_host>_:3893`.
+LDAP_BIND_DN:: Your LDAP bind distinguished name, such as `cn=serviceuser,ou=svcaccts,dc=glauth,dc=com`.
+LDAP_BIND_PASSWORD:: Your LDAP password, such as `mysecret`.
+====
+
+. Create your custom authentication provider backend plugin, that:
+* Provides an authentication provider API.
+* Bridges authentication to your custom authentication provider.
+* Includes an example backend configuration in an `app-config.yaml`file.
++
+--
+.LDAP backend plugin
+====
+.`src/index.ts` wrapper re-exporting the original plugin
+[source,javascript]
+----
+export { default as default } from "@immobiliarelabs/backstage-plugin-ldap-auth-backend";
+----
+
+.`app-config.yaml` backend configuration
+[source,yaml]
+----
+auth:
+  environment: production
+  providers:
+    ldap:
+      production:
+        ldapAuthenticationOptions:
+          userSearchBase: ou=users,dc=glauth,dc=com
+          usernameAttribute: uid
+          adminDn: ${LDAP_BIND_DN}
+          adminPassword: ${LDAP_BIND_PASSWORD}
+          ldapOpts:
+            url:
+              - ${LDAP_URL}
+----
+====
+--
+
+. Create your custom authentication provider frontend plugin, that:
+* Provides a custom `SignInPage` component.
+The `SignInPage` component is the place in a {product-short} app where the frontend API reference is connected to the appropriate backend authentication provider API service.
+* Provides an API client for the backend authentication provider service.
+* Includes an example backend configuration in an `app-config.yaml`file.
++
+--
+.LDAP frontend plugin
+====
+The exported `SignInPage` component delegates to the `LDAPSignInPage` component that is provided by the `backstage-plugin-ldap-auth` package.
+
+.`src/index.ts` wrapper re-exporting the original plugin
+[source,javascript]
+----
+export * from '@immobiliarelabs/backstage-plugin-ldap-auth';
+export { SignInPage } from './components/SignInPage/SignInPage';
+----
+
+.`src/components/SignInPage.ts` `SignInPage` component
+[source,javascript]
+----
+import React from 'react';
+import { SignInPageProps } from '@backstage/core-plugin-api';
+import { LdapAuthFrontendPage } from '@immobiliarelabs/backstage-plugin-ldap-auth';
+
+export function SignInPage(props: SignInPageProps): React.JSX.Element {
+  return <LdapAuthFrontendPage {...props} provider="ldap" />;
+}
+----
+
+.`app-config.yaml` frontend configuration
+[source,yaml]
+----
+dynamicPlugins:
+  frontend:
+    immobiliarelabs-backstage-plugin-ldap-auth:
+      components:
+        - name: SignInPage
+          module: PluginRoot
+          importName: SignInPage
+----
+====
+--
+
+. link:{installing-and-viewing-dynamic-plugins-url}#assembly-third-party-plugins[Export, package, and install your custom plugins].
+. Add your custom plugins configuration to `dynamic-plugins.yaml` in {product-short}.
++
+.`dynamic-plugins.yaml` excerpt for LDAP catalog support
```

Comment:
@gashcrumb Does the `dynamic-plugins.yaml` file content need to be that long? I would assume that if the plugins has defined default values, it is sufficient to enable the plugin. Is it correct?

Reply:
Maybe these LDAP plugins aren't a great example, most of the config here is concerned with getting these plugins to talk to LDAP.  It's probably less important than the touch points where the user is required to integrate either configuration or code to connect authentication providers to the RHDH app.  I have just finished up another example that I could walk you through if you like [here](https://github.com/gashcrumb/dynamic-plugins-multiple-auth-modules).

Reply:
At first sight this example is maybe even more complex, because we have 4 plugins.

Would an example plugin defining static users be sufficient for the demonstration purpose? Or is it a requirement to delegate authentication to an external service?